src/tasks/AotCompilerTask/MonoAOTCompiler.cs

```diff
@@ -305,12 +305,63 @@ public class MonoAOTCompiler : Microsoft.Build.Utilities.Task
     private FileCache? _cache;
     private int _numCompiled;
     private int _totalNumAssemblies;
+    private bool _collectTrimmingEligibleMethodsValueFromPreviousBuild;
 
     private readonly Dictionary<string, string> _symbolNameFixups = new();
     private static readonly char[] s_semicolon = new char[]{ ';' };
 
     private bool ProcessAndValidateArguments()
     {
+        string monoAotPropertyValuesFilePath = Path.Combine(IntermediateOutputPath, "monoAotPropertyValues.txt");
+
+        if (File.Exists(monoAotPropertyValuesFilePath))
+        {
+            string? text = File.ReadAllText(monoAotPropertyValuesFilePath);
+            if (!string.IsNullOrEmpty(text))
+            {
+                Dictionary<string, bool>? properties = JsonSerializer.Deserialize<Dictionary<string, bool>>(text);
+                if ((properties != null) && properties.ContainsKey("CollectTrimmingEligibleMethods"))
+                    if (properties.TryGetValue("CollectTrimmingEligibleMethods", out bool value))
+                        _collectTrimmingEligibleMethodsValueFromPreviousBuild = value;
+            }
+            if (_collectTrimmingEligibleMethodsValueFromPreviousBuild != CollectTrimmingEligibleMethods)
+                File.Delete(monoAotPropertyValuesFilePath);
+        }
+
+        if (!File.Exists(monoAotPropertyValuesFilePath) || ((_collectTrimmingEligibleMethodsValueFromPreviousBuild != CollectTrimmingEligibleMethods)))
+        {
+            Dictionary<string, bool> monoAotPropertyValuesToSave = new();
+            monoAotPropertyValuesToSave.Add("CollectTrimmingEligibleMethods", CollectTrimmingEligibleMethods);
+            string jsonString = JsonSerializer.Serialize(monoAotPropertyValuesToSave);
+            File.WriteAllText(monoAotPropertyValuesFilePath, jsonString);
+            Log.LogMessage(MessageImportance.High, $"Logged the value of CollectTrimmingEligibleMethods in {monoAotPropertyValuesFilePath}");
+        }
+
+        if (_collectTrimmingEligibleMethodsValueFromPreviousBuild != CollectTrimmingEligibleMethods)
+        {
+            DirectoryInfo di = new DirectoryInfo(IntermediateOutputPath);
+            foreach (FileInfo file in di.GetFiles("*.bc"))
+            {
+                Log.LogMessage(MessageImportance.High, $"Deleting {file.Name} to force a new AOT compilation, because the value of CollectTrimmingEligibleMethods has changed");
+                file.Delete();
+            }
+            foreach (FileInfo file in di.GetFiles("*.o"))
```

Comment:
This will get *any* `*.o` files in `obj/`, and not just the ones related to aot.

Reply:
Originally, I didn't intend to remove the `.o` files, as they are not the output of the aot task. However, I ran into a problem where the task (`EmccCompile`) to generate the `.o` files would generated a new `.o` file but would not replace with the old one, as the content is the same. Example,
```
Did not overwrite /Users/yangfan/Documents/work/dotnet_1/runtime/artifacts/obj/mono/Wasm.Console.V8.Sample/wasm/Release/browser-wasm/wasm/for-publish/Wasm.Console.V8.Sample.dll.o as the contents are unchanged
```
That caused a rebuild issue afterwards, because the `.o` files were older than the `.bc` files.
In order to work around this issue, I had to remove the `.o` files here as well.

Maybe, the correct thing to do here is to always overwrite newly generated `.o` files in `EmccCompile`? I am not sure what side effect that might bring.

Reply:
This task should not remove files that were not generated by itself.

Reply:
You are right. Will update `EmccCompile` to always overwrite. Will see if it breaks any existing behavior.

Reply:
The change of `EmccCompile`  triggered `Wasm.Build.NativeRebuild.Tests.FlagsChangeRebuildTests.ExtraEmccFlagsSetButNoRealChange` to fail. We have to use cases which require conflicting changes. What shall we do now?

Reply:
> Originally, I didn't intend to remove the `.o` files, as they are not the output of the aot task. However, I ran into a problem where the task (`EmccCompile`) to generate the `.o` files would generated a new `.o` file but would not replace with the old one, as the content is the same. Example,
> 
> ```
> Did not overwrite /Users/yangfan/Documents/work/dotnet_1/runtime/artifacts/obj/mono/Wasm.Console.V8.Sample/wasm/Release/browser-wasm/wasm/for-publish/Wasm.Console.V8.Sample.dll.o as the contents are unchanged
> ```

that is the correct behavior.

> That caused a rebuild issue afterwards, because the `.o` files were older than the `.bc` files. In order to work around this issue, I had to remove the `.o` files here as well.

Which test was failing because of this?

> Maybe, the correct thing to do here is to always overwrite newly generated `.o` files in `EmccCompile`? I am not sure what side effect that might bring.

That would always trigger rebuilds then. I had solved a similar issue before, but need more details.

Reply:
Discussed offline.